sdk/ai/azure-ai-agents/azure/ai/agents/models/_patch.py

```diff
@@ -445,6 +445,65 @@ def execute(self, tool_call: Any) -> Any:
         :return: The output of the tool operations.
         """
 
+    @staticmethod
+    def _create_tool_resources_from_dict(resources: Dict[str, Any]) -> ToolResources:
+        """
+        Safely converts a dictionary into a ToolResources instance.
+
+        :param resources: A dictionary of tool resources. Should be a mapping
+            accepted by ~azure.ai.agents.models.AzureAISearchToolResource
+        :type resources: Dict[str, Any]
+        :return: A ToolResources instance.
+        :rtype: ToolResources
+        """
+        try:
+            return ToolResources(**resources)
+        except TypeError as e:
+            logger.error("Error creating ToolResources: %s", e)  # pylint: disable=do-not-log-exceptions-if-not-debug
+            raise ValueError("Invalid resources for ToolResources.") from e
+
+    @staticmethod
+    def get_tool_resources(tools: List["Tool"]) -> ToolResources:
+        """
+        Get the tool resources from tools.
+
+        :param tools: The list of tool objects whose resources should be merged.
+        :type tools: List[Tool]
+        :return: A new ``ToolResources`` instance representing the merged view.
+        :rtype: ToolResources
+        """
+        tool_resources: Dict[str, Any] = {}
+        for tool in tools:
+            resources = tool.resources
+            for key, value in resources.items():
+                if key in tool_resources:
+                    # Special handling for MCP resources - they need to be merged into a single list
+                    if isinstance(tool_resources[key], list) and isinstance(value, list):
+                        tool_resources[key].extend(value)
+                    elif isinstance(tool_resources[key], dict) and isinstance(value, dict):
+                        tool_resources[key].update(value)
+                    else:
+                        # For other types, the new value overwrites the old one
+                        tool_resources[key] = value
+                else:
+                    tool_resources[key] = value
+        return Tool._create_tool_resources_from_dict(tool_resources)
+
+    @staticmethod
+    def get_tool_definitions(tools: List["Tool"]) -> List[ToolDefinition]:
```

Comment:
Is this just a helper for `[definition for tool in tools for definition in tool.definitions]`? If so, that should be more efficient than extending a list over and over.